clang/tools/driver/cc1_main.cpp

```diff
@@ -445,20 +445,21 @@ Optional<int> CompileJobCache::tryReplayCachedResult(CompilerInstance &Clang) {
   if (!ResultCacheKey)
     return 1;
 
-  Optional<llvm::cas::ObjectRef> Result;
-  if (auto E = Cache->get(*ResultCacheKey).moveInto(Result))
-    consumeError(std::move(E)); // ignore error and treat it as a cache miss.
-
-  if (Result) {
-    Diags.Report(diag::remark_compile_job_cache_hit)
-        << ResultCacheKey->toString() << CAS->getID(*Result).toString();
-    Optional<int> Status =
-        replayCachedResult(Clang, *Result, /*JustComputedResult=*/false);
-    assert(Status && "Expected a status for a cache hit");
-    return *Status;
-  }
-  Diags.Report(diag::remark_compile_job_cache_miss)
-      << ResultCacheKey->toString();
+  if (Expected<Optional<llvm::cas::ObjectRef>> Result =
+          Cache->get(*ResultCacheKey)) {
+    if (*Result) {
+      Diags.Report(diag::remark_compile_job_cache_hit)
+          << ResultCacheKey->toString() << CAS->getID(**Result).toString();
+      Optional<int> Status =
+          replayCachedResult(Clang, **Result, /*JustComputedResult=*/false);
+      assert(Status && "Expected a status for a cache hit");
+      return *Status;
+    }
+    Diags.Report(diag::remark_compile_job_cache_miss)
+        << ResultCacheKey->toString();
+  } else
+    Diags.Report(diag::err_compile_job_cache_result_not_found)
+        << toString(Result.takeError());
```

Comment:
Shouldn't this be followed with `return 1;`?

Reply:
This is a fatal error but sure we can return 1 just to be clean.